### PR TITLE
docs(self-managed): add upgrade steps before helm chart v8.0.13

### PR DESCRIPTION
**Pull request by aabouzaid**
## What is the purpose of the change

Added upgrade steps before Helm chart `v8.0.13`, which requires moving the upgrade to its own page. 

## Are there related marketing activities

No.

## When should this change go live?

Before the Camunda Platform `8.1.0` release.

## PR Checklist

- [x] My changes are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory.
- [x] My changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer.

### CODE REVIEW

File: docs/self-managed/platform-deployment/helm-kubernetes/upgrade.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+Error: UPGRADE FAILED: execution error at (camunda-platform/charts/identity/templates/tasklist-secret.yaml:10:22):
+PASSWORDS ERROR: You must provide your current passwords when upgrading the release.
+                 Note that even after reinstallation, old credentials may be needed as they may be kept in persistent volume claims.
+                 Further information can be obtained at https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases
+
+    'global.identity.auth.tasklist.existingSecret' must not be empty, please add '--set global.identity.auth.tasklist.existingSecret=$TASKLIST_SECRET' to the command. To get the current value:
+
+        export TASKLIST_SECRET=$(kubectl get secret --namespace "zell-c8-optimize" "camunda-platform-test-tasklist-identity-secret" -o jsonpath="{.data.tasklist-secret}" | base64 --decode)
+```
```
Comment (akeller) [inline docs/self-managed/platform-deployment/helm-kubernetes/upgrade.md:36]: Is the spacing here intentional? Particularly in lines 33-35?

And I'm assuming the first line (L26) is the command you run before seeing this error. Is this also included (echoed) in the return? If this is the command the user enters, let's indicate it with a `$`.

Comment (aabouzaid) [reply docs/self-managed/platform-deployment/helm-kubernetes/upgrade.md]: I admit it looks weird :sweat_smile: 
but the whole section is the output of the first line:
```
helm upgrade camunda-platform-test camunda/camunda-platform
```

I've split it to make it more clear which is the command and which is the output.

File: sidebars.js [hunk 2]
```diff
@@ -634,7 +634,8 @@ module.exports = {
         {
           "Helm/Kubernetes": [
             "self-managed/platform-deployment/helm-kubernetes/overview",
-            "self-managed/platform-deployment/helm-kubernetes/deployment",
+            "self-managed/platform-deployment/helm-kubernetes/deploy",
+            "self-managed/platform-deployment/helm-kubernetes/upgrade",
```
Comment (pepopowitz) [inline sidebars.js:638]: Due to the fact that we now have separate versions for Optimize than the rest of the docs, parallel sidebar changes will need to be made at https://github.com/camunda/camunda-platform-docs/blob/main/optimize_sidebars.js#L1153-L1156.

**Review by pepopowitz — CHANGES_REQUESTED** [review]
Additional changes are required to keep the sidebars from drifting in two separate areas of the site. 

See #1116 for more details on why these changes are needed, or let me know if I can further clarify!

File: sidebars.js (hunk 2 above)
Comment (akeller) [reply sidebars.js:638]: I took care of this in the latest commit.

**Review by akeller — APPROVED** [review]

**Pull request merged**